### PR TITLE
feat(concerto v1) : update

### PR DESCRIPTION
**Pull request by dselman**
Upgrades the build script and models to support Concerto v1.

### Changes
- New model files for those that depend on `org.accordproject.cicero.runtime`
- Added support for reading a compatibility comment within the model file `// requires: concerto-core:>= 1.0.0-20210215133932` to control whether the model file gets processed or not, based on the version of `concerto-core` listed in package.json

### CODE REVIEW

**Comment by dselman** [comment]
Note that we need `import org.accordproject.cicero.runtime.Request from https://models.accordproject.org/cicero/runtime@0.2.0.cto` to get published...

File: build.js [hunk 1]
```diff
@@ -196,14 +197,32 @@ const rootDir = resolve(__dirname, './src');
...
+    let versionRange = null;
+    while ((m = regex.exec(modelText)) !== null) {
+        if (m.index === regex.lastIndex) {
+            regex.lastIndex++;
+        }
+        if(m && m.length > 1) {
+            versionRange = m[1];
+        }
+    }
+    return versionRange ? semver.satisfies( concertoVersion, versionRange ) : true;
```
Comment (mttrbrts) [inline build.js:218]: To be clear, if there is no explicit version range comment, then the model is assumed to be compatible.

File: build.js [hunk 2]
```diff
@@ -196,14 +197,32 @@ const rootDir = resolve(__dirname, './src');
...
+    let m;
+    let versionRange = null;
+    while ((m = regex.exec(modelText)) !== null) {
+        if (m.index === regex.lastIndex) {
+            regex.lastIndex++;
+        }
+        if(m && m.length > 1) {
+            versionRange = m[1];
+        }
+    }
```
Comment (mttrbrts) [inline build.js:217]: Can this be replaced with this ?
```js
const regex = /^\/\/.*requires:.*concerto-core:(?<versionRange>.*)$/m;
const { versionRange } = modelText.match(regex).groups;
```

File: build.js [hunk 3]
```diff
@@ -218,74 +237,84 @@ let modelFileIndex = [];
...
+            const modelFile  = new ModelFile(modelManager, modelText, file);     
+            console.log(`Processing ${modelFile.getNamespace()}`);
+            let modelFilePlantUML = '';
+            // passed validation, so copy to build dir
+            const dest = file.replace('/src/', '/build/');
+            const destPath = path.dirname(dest);
+            const relative = destPath.slice(buildDir.length);
+            // console.log('dest: ' + dest);
+            // console.log('destPath: ' + destPath);
+            // console.log('relative: ' + relative);
```
Comment (mttrbrts) [inline build.js:259]: Please remove these commented lines.

File: build.js [hunk 4]
```diff
@@ -218,74 +237,84 @@ let modelFileIndex = [];
...
+                if (!isLegacyModelVersionScheme) { // Skip indexing models with the old versioning scheme, they have all been migrated now
+                    const semverStr = modelFile.getName().split("@").pop().slice(0,-4);
+                    const isSemverVersionScheme = semver.valid(semverStr);
+                    const modelVersion = isSemverVersionScheme ? `(${semverStr})` : '(0.1.0)';
+    
+                    modelManager.addModelFile(modelFile, modelFile.getName(), true);
+    
+                    // use the FORCE_PUBLISH flag to disable download of
+                    // external models and model validation
+                    if(!process.env.FORCE_PUBLISH) {
```
Comment (mttrbrts) [inline build.js:279]: Is `FORCE_PUBLISH` new? Should it be a parameter of `updateExternalModels`?

Comment (dselman) [reply build.js:279]: No, it's not new -- we always had this env var for the script.

**Review by mttrbrts — APPROVED** [review]

**Comment by jeromesimeon** [comment]
This shouldn't have gone into master. I worry we may have to revert this PR.

**Comment by jeromesimeon** [comment]
Why do we need a new runtime model?

**Comment by jeromesimeon** [comment]
The newly proposed base AP models for Concerto v1 are here and will need to be reviewed and checked. Those are meant to be consistent with the Ergo compiler v1. https://github.com/accordproject/models/tree/release-1.0/src/accordproject

**Pull request merged**